op-node/rollup/derive/channel_compressor.go

```diff
@@ -0,0 +1,85 @@
+package derive
+
+import (
+	"bytes"
+	"compress/zlib"
+	"fmt"
+	"io"
+
+	"github.com/andybalholm/brotli"
+)
+
+const (
+	ChannelVersionBrotli byte = 0x01
+)
+
+type CompressorWriter interface {
+	Write([]byte) (int, error)
+	Flush() error
+	Close() error
+	Reset(io.Writer)
+}
+
+type ChannelCompressor struct {
+	writer          CompressorWriter
+	compressionAlgo CompressionAlgo
+	compressed      *bytes.Buffer
+}
+
+func NewChannelCompressor(algo CompressionAlgo) (*ChannelCompressor, error) {
```

Comment:
making a ChannelCompressor a struct, and NewChannelCompressor return a pointer to a struct (which may behave one of two concrete ways), makes me think that this ChannelCompressor should actually be an interface, with two concrete implementations. Then this factory function would look at the Algo and build the correct concrete implementation. This would allow you to make the Brotli `Reset` function special purpose from the Zlib version.